Makefile

```diff
@@ -463,17 +474,21 @@ endif
 bootstrap: coreboot
 	$(MAKE) all
 
-# Compile everything the first time
+# Old targets for compiling everything for the first time
 
 .PHONY: world
-world: coldstart
+world: boot/initialised
 	$(MAKE) all
 
 # Compile also native code compiler and libraries, fast
 .PHONY: world.opt
 world.opt: checknative
 	$(MAKE) coldstart
 	$(MAKE) opt.opt
+	@echo
+	@echo "$(MAKE) world.opt is deprecated; use $(MAKE) all \
+or even just $(MAKE)"
+	@echo "To rebuild the .opt tools, use $(MAKE) opt.opt"
```

Comment:
As a relative novice of the compiler's build system, this is the sort of message that would annoy me: "why are they breaking my workflow now?". I am probably guilty of using `make world.opt` out of habit without understanding the subtleties, and I would be happy to learn that I have been doing something wrong. So the message could be the occasion to be more pedagogical, in fact it could be a simple advice where the word "deprecated" doesn't even need to appear (I find it a bit patronizing: "we have decided for you, and it is wrong not to do as we say").

Reply:
I'm not sure which version of "now" you mean here - it's not being broken _now_, it's telling you that it may be broken in the future! The word deprecated is used pervasively - I feel you may need to find it less patronizing, if you'll inevitable excuse the meta-pun in that!

I softened the message slightly by re-wording "_you can now_ use".

Reply:
Thank you for the rephrasing which is a bit better. It does not really address the point I was trying to make though. Now I am even more confused, even after reading again the full PR and discussions: first you say that the deprecation "should encourage less surprising core developer practices" (which? not sure I understood), but now you say that it may be broken in the future (how?). I am a user who does not have stakes in world.opt, and I am open-minded about learning better practices, but for now, I am not convinced by the the change. If you want to deprecate something, you first should be able to communicate clearly to users why it is being deprecated. (I do not think I understood what you meant in the second sentence either.)

Reply:
I’m sorry for the confusion here, but I haven’t changed anything - deprecating means stop using it because it not be there at some point in the future. I have no idea if/when that would actually happen.

Reply:
As to why it's an improvement, the fact you don't know why sort of makes the point. At present, repeatedly running `make world.opt` will put a new runtime and standard library in `boot/` (that's the job of the `coldstart` target). If you have made changes which will ultimately require a bootstrap, you've now masked that. If you run the new `make all`, you'll know because your compiler won't build with the *original* one in `boot/` which was assembled the first time you ran `make all`.

Reply:
You have no idea if/when the target world.opt would be changed or removed, so you are literally telling us that it is not deprecated.

Thanks for the explanation. I was already trusting that it is an improvement, otherwise you would not be doing it. But you have only just explained it to me now.

I repeat my advice of providing a pedagogical message instead. In fact, I think you could get your message across to users simply by providing a (usefully!) informative message, along the following lines:

* When initializing for the first time: "boot/ has been initialized; you can now use make all or make opt.opt to compile without rebuilding it."
* When rebuilding boot/: "boot/ has been rebuilt."

Reply:
Such an informative message could be appropriate for make world too.